mantidproject/mantid
# Fix special values propagating in SumSpectra

Description of work.
This PR fixes an issue with SumSpectra where any spectra with "special" values such as NaN would cause the final single spectra to have a `NaN` value(s). This is most apparent when summing masked out detectors whose values have been set to `NaN`. 
We now call replace special values before processing the workspace to replace said values with 0. This was already done within `rebinnedOutput` so the step was moved into its own method and called for 2D workspaces. 

Additionally this PR contains another fix to stop the GUI attempting to parse a path to an existing calibration on startup. Previously a warning would appear when opening the GUI completely unrelated to the GUI start-up.

**To test:**

Download the following .nxs file or create a workspace with random `nan` values in the data:
[18123_Engg_masked_ws.zip](https://github.com/mantidproject/mantid/files/617056/18123_Engg_masked_ws.zip)
Run SumSpectra 
Check the data - any `nan` values should contribute 0 and the final workspace should not contain any special floating point values 

Fixes #18123 .

[Release Notes](https://github.com/mantidproject/mantid/pull/18128/files#diff-f7793b747fb7ac7103dcdcb26c5d7ca5)


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
